src/maxminddb.c

```diff
@@ -1384,7 +1384,10 @@ LOCAL int decode_one(MMDB_s *mmdb, uint32_t offset,
 {
     const uint8_t *mem = mmdb->data_section;
 
-    if (offset + 1 > mmdb->data_section_size) {
+    // We subtract rather than add as it possible that offset + 1
+    // could overflow for a corrupt database while an underflow
+    // from data_section_size - 1 should not be possible.
+    if (offset > mmdb->data_section_size - 1) {
```

Comment:
I don't think it is possible to construct such a database as the metadata will be more than three bytes. I did the additional check on values that could be significantly larger based on their construction. However, I am not opposed to adding a check when we open the database.